src/use_factor_source/derive_keys_with_factor_source_driver.rs

```diff
@@ -0,0 +1,30 @@
+use crate::prelude::*;
+
+pub type BatchDerivePublicKeysResponse = BatchUseFactorSourceResponse<DeriveKeyID, HDPublicKey>;
+
+pub type BatchDerivePublicKeysRequest = BatchUseFactorSourceRequest<DeriveKeyID, DerivationPath>;
+
+#[async_trait::async_trait]
+pub trait DeriveKeysWithFactorSourceDriver:
+    UseFactorSourceDriver<(), DerivationPath, HDPublicKey>
+{
+    /// Derives many keys from many factor sources for many entities.
+    async fn batch_derive_public_keys(
+        &self,
+        request: BatchDerivePublicKeysRequest,
+    ) -> Result<BatchDerivePublicKeysResponse>;
```

Comment:
Deriving of keys is not skippable! So we return `Result< BatchDerivePublicKeysResponse >` here instead of `Result<UseFactorsAction< BatchDerivePublicKeysResponse>>`  (which is the API of `UseFactorSourceDriver` (and `SignWithFactorSourceDriver`)).

If you can come up with a better way of organising this, please tell me! :)